hathor/manager.py

```diff
@@ -1013,16 +1029,18 @@ def on_new_tx(self, tx: BaseTransaction, *, conn: Optional[HathorProtocol] = Non
             tx.update_initial_metadata()
             self.tx_storage.save_transaction(tx)
 
-        if tx.is_transaction:
+        if tx.is_transaction and self._enable_sync_v2:
+            assert self.tx_storage.indexes.deps is not None
             self.tx_storage.indexes.deps.remove_from_needed_index(tx.hash)
 
-        try:
-            self.step_validations([tx])
-        except (AssertionError, HathorError) as e:
-            if not fails_silently:
-                raise InvalidNewTransaction('step validations failed') from e
-            self.log.warn('on_new_tx(): step validations failed', tx=tx.hash_hex, exc_info=True)
-            return False
+        if self._enable_sync_v2:
+            try:
+                self.step_validations([tx])
```

Comment:
Should we rename it to `self.sync_v2_step_validations(...)`?

Reply:
That makes sense.

Reply:
Done.

Reply:
Thanks!